JetStream cluster could not decode consumer snapshot

## Defect

Make sure that these boxes are checked before submitting your issue -- thank you!

 - [ ] Included `nats-server -DV` output
 - [ ] Included a [Minimal, Complete, and Verifiable example] (https://stackoverflow.com/help/mcve)

#### Versions of `nats-server` and affected client libraries used:
2.6.5

#### OS/Container environment:
alpine 

#### Steps or code to reproduce the issue:
- stop 1 of 3 pods
- start 1 of 3 pods

#### Expected result:
- correct start

#### Actual result:
- incorrect start

```
kubectl -n production  logs collector-event-nats-2 -c nats
[87] 2021/12/01 17:08:48.861842 [INF] Starting nats-server
[87] 2021/12/01 17:08:48.861966 [INF]   Version:  2.6.5
[87] 2021/12/01 17:08:48.861971 [INF]   Git:      [ea48105]
[87] 2021/12/01 17:08:48.861974 [INF]   Name:     collector-event-nats-2
[87] 2021/12/01 17:08:48.861989 [INF]   Node:     dcRC6Zf4
[87] 2021/12/01 17:08:48.861992 [INF]   ID:       NBPSM6UJMEPOKF6FNJODGZX2YDULV3L7NVNK4RNXECA72OE467G27VDD
[87] 2021/12/01 17:08:48.861998 [INF] Using configuration file: /etc/nats-config/nats.conf
[87] 2021/12/01 17:08:48.863486 [INF] Starting JetStream
[87] 2021/12/01 17:08:48.863829 [INF]     _ ___ _____ ___ _____ ___ ___   _   __  __
[87] 2021/12/01 17:08:48.863846 [INF]  _ | | __|_   _/ __|_   _| _ \ __| /_\ |  \/  |
[87] 2021/12/01 17:08:48.863848 [INF] | || | _|  | | \__ \ | | |   / _| / _ \| |\/| |
[87] 2021/12/01 17:08:48.863850 [INF]  \__/|___| |_| |___/ |_| |_|_\___/_/ \_\_|  |_|
[87] 2021/12/01 17:08:48.863851 [INF] 
[87] 2021/12/01 17:08:48.863853 [INF]          https://docs.nats.io/jetstream
[87] 2021/12/01 17:08:48.863854 [INF] 
[87] 2021/12/01 17:08:48.863856 [INF] ---------------- JETSTREAM ----------------
[87] 2021/12/01 17:08:48.863869 [INF]   Max Memory:      512.00 MB
[87] 2021/12/01 17:08:48.863872 [INF]   Max Storage:     50.00 GB
[87] 2021/12/01 17:08:48.863875 [INF]   Store Directory: "/data/jetstream"
[87] 2021/12/01 17:08:48.863876 [INF] -------------------------------------------
[87] 2021/12/01 17:08:48.870153 [INF]   Restored 0 messages for stream "error-collector"
[87] 2021/12/01 17:08:56.299199 [INF]   Restored 5,019,337 messages for stream "event-collector"
[87] 2021/12/01 17:09:08.709647 [INF]   Restored 5,184,221 messages for stream "request-collector"
[87] 2021/12/01 17:09:08.709714 [INF]   Recovering 1 consumers for stream - "error-collector"
[87] 2021/12/01 17:09:08.710007 [INF]   Recovering 1 consumers for stream - "event-collector"
[87] 2021/12/01 17:09:08.710163 [INF]   Recovering 1 consumers for stream - "request-collector"
[87] 2021/12/01 17:09:08.710383 [INF] Starting JetStream cluster
[87] 2021/12/01 17:09:08.710387 [INF] Creating JetStream metadata controller
[87] 2021/12/01 17:09:08.710598 [INF] JetStream cluster recovering state
[87] 2021/12/01 17:09:08.712197 [INF] Starting http monitor on 0.0.0.0:8222
[87] 2021/12/01 17:09:08.712280 [INF] Listening for client connections on 0.0.0.0:4222
[87] 2021/12/01 17:09:08.712503 [INF] Server is ready
[87] 2021/12/01 17:09:08.712522 [INF] Cluster name is nats
[87] 2021/12/01 17:09:08.712548 [INF] Listening for route connections on 0.0.0.0:6222
[87] 2021/12/01 17:09:08.716585 [INF] 10.244.243.194:6222 - rid:22 - Route connection created
[87] 2021/12/01 17:09:08.716643 [INF] 10.244.69.177:6222 - rid:23 - Route connection created
[87] 2021/12/01 17:09:08.753549 [WRN] Error applying entries to '$G > event-collector': last sequence mismatch
[87] 2021/12/01 17:09:08.768254 [WRN] Resetting stream cluster state for '$G > event-collector'
[87] 2021/12/01 17:09:08.795096 [WRN] Error applying entries to '$G > event-collector': last sequence mismatch
[87] 2021/12/01 17:09:08.797944 [ERR] JetStream cluster could not decode consumer snapshot for '$G > request-collector > request-collector-1' [C-R3F-XO5nmQ6F]
panic: corrupt state file

goroutine 179 [running]:
github.com/nats-io/nats-server/server.(*jetStream).applyConsumerEntries(0xc000162160, 0xc000625500, 0xc00059e4c0, 0x0, 0x5, 0x1)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:3037 +0xe6f
github.com/nats-io/nats-server/server.(*jetStream).monitorConsumer(0xc000162160, 0xc000625500, 0xc00067eab0)
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:3006 +0x68f
github.com/nats-io/nats-server/server.(*jetStream).processClusterCreateConsumer.func1()
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:2800 +0x3c
created by github.com/nats-io/nats-server/server.(*Server).startGoRoutine
        /home/travis/gopath/src/github.com/nats-io/nats-server/server/server.go:2867 +0xc5

```

--- a/server/filestore.go
+++ b/server/filestore.go
@@ -5348,11 +5348,14 @@ func decodeConsumerState(buf []byte) (*ConsumerState, error) {
 				dseq = readSeq()
 			}
 			ts := readTimeStamp()
-			if sseq == 0 || ts == -1 {
+			if ts == -1 {
 				return nil, errCorruptState
 			}
 			// Adjust seq back.
 			sseq += state.AckFloor.Stream
+			if sseq == 0 {
+				return nil, errCorruptState
+			}
 			if version == 2 {
 				dseq += state.AckFloor.Consumer
 			}

--- a/server/filestore_test.go
+++ b/server/filestore_test.go
@@ -2012,6 +2012,44 @@ func TestFileStoreConsumerEncodeDecodeRedelivered(t *testing.T) {
 	}
 }
 
+func TestFileStoreConsumerEncodeDecodePendingBelowStreamAckFloor(t *testing.T) {
+	state := &ConsumerState{}
+
+	state.Delivered.Consumer = 1192
+	state.Delivered.Stream = 10185
+	state.AckFloor.Consumer = 1189
+	state.AckFloor.Stream = 10815
+
+	now := time.Now().Round(time.Second).Add(-10 * time.Second).UnixNano()
+	state.Pending = map[uint64]*Pending{
+		10782: {1190, now},
+		10810: {1191, now + int64(time.Second)},
+		10815: {1192, now + int64(2*time.Second)},
+	}
+	buf := encodeConsumerState(state)
+
+	rstate, err := decodeConsumerState(buf)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	if len(rstate.Pending) != 3 {
+		t.Fatalf("Invalid pending: %v", rstate.Pending)
+	}
+	for k, v := range state.Pending {
+		rv, ok := rstate.Pending[k]
+		if !ok {
+			t.Fatalf("Did not find sseq=%v", k)
+		}
+		if !reflect.DeepEqual(v, rv) {
+			t.Fatalf("Pending for sseq=%v should be %+v, got %+v", k, v, rv)
+		}
+	}
+	state.Pending, rstate.Pending = nil, nil
+	if !reflect.DeepEqual(*state, *rstate) {
+		t.Fatalf("States do not match: %+v vs %+v", state, rstate)
+	}
+}
+
 func TestFileStoreWriteFailures(t *testing.T) {
 	// This test should be run inside an environment where this directory
 	// has a limited size.

